specutils/manipulation/extract_spectral_region.py

```diff
@@ -87,7 +99,19 @@ def _subregion_to_edge_pixels(subregion, spectrum):
         if not spectral_axis.unit.is_equivalent(u.pix):
             right_index = ceil(subregion[1].value)
         else:
-            right_index = int(np.floor(right_func(subregion).value)) + 1
+            if order != "ascending":
+                raise ValueError("u.pix handling should always be ascending")
```

Comment:
```suggestion
                raise ValueError("u.pix spectral axes should always be ascending")
```

Same comment as above.